packages/protocol/contracts/common/MetaTransactionWalletDeployer.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity ^0.5.13;
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+
+import "./interfaces/ICeloVersionedContract.sol";
+import "./interfaces/IMetaTransactionWalletDeployer.sol";
+import "./proxies/MetaTransactionWalletProxy.sol";
+import "./ExternalCall.sol";
+import "./Initializable.sol";
+import "./MetaTransactionWallet.sol";
+
+contract MetaTransactionWalletDeployer is
+  IMetaTransactionWalletDeployer,
+  ICeloVersionedContract,
+  Initializable,
+  Ownable
+{
+  using SafeMath for uint256;
+  using BytesLib for bytes;
+
+  mapping(address => address) public wallets;
+  mapping(address => bool) public canDeploy;
+
+  event WalletDeployed(address indexed owner, address indexed wallet, address implementation);
+  event DeployerStatusGranted(address indexed addr);
+  event DeployerStatusRevoked(address indexed addr);
+
+  /**
+     * @dev Verifies that the sender is allowed to deploy a wallet
+     */
+  modifier onlyCanDeploy() {
+    require(msg.sender == owner() || canDeploy[msg.sender] == true, "not-allowed");
```

Comment:
I'm unfamiliar with any conventions clabs has on revert messages, but if I were debugging I might appreciate something more specific like: "sender not authorized to deploy wallet". "not-allowed", if used elsewhere too, could make debugging difficult.